lizard_auth_server/views_api.py

```diff
@@ -300,3 +300,121 @@ def get_organisations(self, portal):
             'organisations': [
                 organisation.as_dict()
                 for organisation in models.Organisation.objects.all()]}
+
+
+class GetRolesView(FormView):
+    """
+    View that can be used to get
+    """
+    # return JsonResponse({"message": "Ik ben een apennootjes-eetfabriek, Jk."})
+
+    form_class = forms.DecryptForm
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super(GetRolesView, self).dispatch(
+            request, *args, **kwargs)
+
+    @method_decorator(never_cache)
+    def post(self, request, *args, **kwargs):
+        return super(GetRolesView, self).post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        return JsonResponse(self.get_roles(form.portal))
+
+    def form_invalid(self, form):
+        logger.error(
+            'Error while decrypting roles form: {}'.format(form.errors.as_text())
+        )
+        return HttpResponseBadRequest('Bad signature')
+
+    def get_roles(self, portal):
+        roles = [
+            role.as_dict()
+            for role in models.Role.objects.all()
+            if role.portal == portal]
+        # roles = 'rolletjes!!!'
+        return {
+            'success': True,
```

Comment:
Waarom wordt het op deze manier doorgegeven? Bij form_invalid zie ik geen dict terugkomen met `success=False`. Dus als er een "200 OK" terugkomt is dat al een teken dat het lukte. En waarom is de message nodig? De server wil daarmee zeggen dat de client alles goed ontvangen heeft? Dat kan de client zelf toch wel loggen?

=> Nog even kijken. Teveel success=True dingen in een response, dat gaat de richting van SOAP op in plaats van REST.